front/lib/api/assistant/configuration.ts

```diff
@@ -1504,6 +1509,19 @@ async function _createAgentDataSourcesConfigData(
   const results = await Promise.all(getDataSourcesQueries);
   const dataSources = results.flat();
 
+  // Since the UI does not currently provide the data source view,
+  // we retrieve the view associated with the data from the global vault
+  // and assign it to the agent data source configuration.
+  const uniqueDataSources = _.uniqBy(dataSources, (ds) => ds.id);
+  const globalVault = await VaultResource.fetchWorkspaceGlobalVault(auth);
+  const dataSourceViews =
+    await DataSourceViewResource.listForDataSourcesInVault(
+      auth,
```

Comment:
Open a thread [here](https://dust4ai.slack.com/archives/C050SM8NSPK/p1722357112317509) to understand why we can have several workspace ids.

Reply:
Responded there 👍